core/compilers.ts

```diff
@@ -4,7 +4,10 @@ import { TableContext } from "df/core/table";
 import * as utils from "df/core/utils";
 import { SyntaxTreeNode, SyntaxTreeNodeType } from "df/sqlx/lexer";
 
-export function compile(code: string, path: string) {
+export function compile(code: string, path?: string) {
+  if (!path) {
```

Comment:
Can we just call the correct method from the caller? Changing the logic based on the presence of path or not is a bit strange to me